packages/presentation/lib/ui/splash_screen/bloc/splash_bloc.dart

```diff
@@ -1,33 +1,81 @@
+import 'dart:io';
+
 import 'package:domain/use_case/splash_duration_use_case.dart';
+import 'package:domain/use_case/check_version_use_case.dart';
 import 'package:presentation/base/bloc.dart';
+import 'package:presentation/generated/l10n.dart';
+import 'package:presentation/library/const/links.dart';
+import 'package:presentation/navigation/base_arguments.dart';
 import 'package:presentation/ui/movie_page/movie_widget.dart';
+import 'package:presentation/ui/splash_screen/bloc/splash_tile.dart';
+import 'package:url_launcher/url_launcher.dart';
 
-abstract class SplashBloc extends Bloc {
+abstract class SplashBloc extends Bloc<BaseArguments, SplashTile> {
   factory SplashBloc(
     SplashDurationUseCase durationUseCase,
+    CheckVersionUseCase versionUseCase,
   ) =>
       SplashBlocImpl(
         durationUseCase,
+        versionUseCase,
       );
+
+  Future<void> updateTap();
 }
 
-class SplashBlocImpl extends BlocImpl implements SplashBloc {
+class SplashBlocImpl extends BlocImpl<BaseArguments, SplashTile>
+    implements SplashBloc {
+  var _tile = SplashTile.init();
   final SplashDurationUseCase _durationUseCase;
+  final CheckVersionUseCase _versionUseCase;
 
   SplashBlocImpl(
     this._durationUseCase,
+    this._versionUseCase,
   );
 
   @override
-  void initState() {
+  void initState() async {
     super.initState();
-    nextScreen();
+    final versionResult = await checkVersion();
+    nextScreen(versionResult);
+  }
+
+  Future<String> checkVersion() async {
```

Comment:
actually doesn't seem pretty. check for the empty string is quite complicated from the logic side. imho, better solution is to wrap usecase and nextscreen call into try/catch block and throw an exception in case version is not suitable. than in catch block handle the exception

Reply:
I changed the logic for showing alerts and got rid of the usecase duration. Now only when pressing "later" we will go to the next screen, in this case, try/catch would be redundant and inconvenient